depthmapX/GraphDoc.cpp

```diff
@@ -312,6 +312,74 @@ void QGraphDoc::ProcPostMessage(int m, int x, int y)
     }
 }
 
+void QGraphDoc::OnVGALinksFileImport()
+{
+    if (m_communicator) {
+        return; // Locked
+    }
+
+    // change the view before loading the file to make the changes apparent
+    ((QDepthmapView*)m_view[VIEW_MAP])->m_showlinks = true;
+    SetRedrawFlag(VIEW_MAP,REDRAW_POINTS, NEW_DEPTHMAPVIEW_SETUP);
+
+    QString template_string;
+    template_string += "Text files (*.txt *.csv)\n";
+    template_string += "All files (*.*)";
+
+    QFileDialog::Options options = 0;
+    QString selectedFilter;
+    QStringList infiles = QFileDialog::getOpenFileNames(
+                                0, tr("Import Links File"),
+                                "",
+                                template_string,
+                                &selectedFilter,
+                                options);
+
+    if (!infiles.size())
+    {
+        // no file selected
+        return;
+    }
+    if (infiles.size() > 1)
+    {
+        QMessageBox::warning(this, tr("Warning"), tr("Multiple files selected!"),
+                             QMessageBox::Ok, QMessageBox::Ok);
+        return;
+    }
+
+    std::string fileName = infiles[0].toStdString();
+
+    ifstream fileStream(fileName);
+    if (fileStream.fail())
+    {
+      QMessageBox::warning(this, tr("Warning"), tr("Unable to read text file.\nPlease check that another program is not using it."),
+                           QMessageBox::Ok, QMessageBox::Ok);
+      return;
+    }
+    else
+    {
+        try
+        {
+            PointMap& currentMap = m_meta_graph->getDisplayedPointMap();
+            vector<PixelRefPair> newLinks = depthmapX::pixelateMergeLines(
+                        EntityParsing::parseLines(fileStream, '\t'), currentMap);
+            depthmapX::mergePixelPairs(newLinks, currentMap);
+            SetRedrawFlag(VIEW_MAP,REDRAW_POINTS, NEW_DEPTHMAPVIEW_SETUP);
+        }
+        catch (depthmapX::BaseException& e)

```

Comment:
Uh oh. I think there's too much in the try block - basically the user will want to know whether reading/converting/pixelating the input failed (fine to carry on) or mergePixelPairs has blown up (not fine, the actual map might be corrupted).